ui/ui-components/__tests__/components/setupStepsNavProgressBar.tsx

```diff
@@ -0,0 +1,127 @@
+import { render, screen } from "@testing-library/react";
+import Component from "../../components/navigation/setupStepsNavProgressBar";
+import mockAxios from "jest-mock-axios";
+import { Actions } from "../../utils/apiData";
+import { useApi } from "../../hooks/useApi";
+
+const useRouter = jest.spyOn(require("next/router"), "useRouter");
+const mockSetupStepHandler = { subscribe: () => {}, unsubscribe: () => {} };
+
+describe("setupStepsNavProgressBar", () => {
+  describe("hidden", () => {
+    beforeEach(() => {
+      useRouter.mockImplementation(() => ({
+        pathname: "/",
+        asPath: "/",
+      }));
+    });
+
+    for (const page of ["/", "/session/sign-in"]) {
+      it(`does not render on ${page}`, () => {
+        useRouter.mockImplementation(() => ({ pathname: page, asPath: page }));
+        render(
+          <Component
+            execApi={() => {}}
+            setupStepHandler={mockSetupStepHandler}
+          />
+        );
+        expect(screen.queryByTestId("setupStepsProgressBar")).toBeNull();
+      });
+    }
+  });
+
+  describe("on a logged in page", () => {
+    const { execApi } = useApi({});
+
+    beforeEach(() => {
+      useRouter.mockImplementation(() => ({
+        pathname: "/dashboard",
+        asPath: "/dashboard",
+        prefetch: async () => {},
+      }));
+    });
+
+    afterEach(() => {
+      mockAxios.reset();
+    });
+
+    it("doesn't display without steps", async () => {
+      render(
+        <Component execApi={execApi} setupStepHandler={mockSetupStepHandler} />
+      );
+
+      expect(screen.queryByTestId("setupStepsProgressBar")).toBeNull();
+    });
+
+    it("displays with steps", async () => {
+      let stepsResponse: Actions.SetupStepsList = {
```

Comment:
Use the Action types to be sure that your mock request data is formatted properly.